Makefile

```diff
@@ -1,23 +1,54 @@
-PYTHON=python3
-
-.PHONY: deps
-deps:
-	$(PYTHON) -m pip install --upgrade pip
-	$(PYTHON) -m pip install --upgrade -r requirements-hacking.txt
+SRC_DIRS = rebasebot tests
 
 .PHONY: unittests
-unittests:
+unittests: ## Run unit & integration tests
 	hack/tests.sh
 
 .PHONY: lint
-lint:
-	$(PYTHON) -m flake8 --max-line-length=120 rebasebot tests
-	$(PYTHON) -m pylint rebasebot tests
-	$(PYTHON) -m mypy rebasebot tests --no-strict-optional --ignore-missing-imports
+lint: ## Run lint and format in check mode
+	uv tool run ruff check $(SRC_DIRS)
+	uv tool run ruff format --check $(SRC_DIRS)
+
+.PHONY: lint-fix
+lint-fix: ## Fix fixable lint issues and format code
+	uv tool run ruff format $(SRC_DIRS)
+	uv tool run ruff check --fix $(SRC_DIRS)
 
 .PHONY: venv
-venv:
-	$(PYTHON) -m venv env
+venv: ## Create venv
+	uv venv
+
+.PHONY: install
+install: ## Install into your user python environment.
+	python -m pip install --user .
 
-install:
-	$(PYTHON) -m pip install --user .
+.PHONY: build
+build: ## Create build tarball
+	uv build
+
+.PHONY: help
+help: ## Display this help screen
+	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-20s\033[0m %s\n", $$1, $$2}'
+
+.PHONY: clean
+clean:
+	rm -r dist/ build/ .pytest_cache/ .mypy_cache rebasebot.egg-info .coverage
+
+# old target still referenced in ci
+.PHONY: deps
+deps:
+	# Debug information for permission issues
+	echo "=== Debug Information ==="
+	echo "Current working directory: $(pwd)"
+	echo "Current user: $(whoami)"
+	echo "Current user ID: $(id)"
```

Comment:
IIRC you need to use double dollars in Make to escape shell dollars. e.g. `$$(pwd)`